lib/Onyx.js

```diff
@@ -1025,17 +1028,22 @@ function initializeWithDefaultKeyStates() {
  * Storage.setItem() from Onyx.clear() will have already finished and the merged
  * value will be saved to storage after the default value.
  *
+ * @param {Array} keysToPreserve ONYXKEYS which values you want to keep
  * @returns {Promise<void>}
  */
-function clear() {
+function clear(keysToPreserve = []) {
     return getAllKeys()
         .then((keys) => {
-            _.each(keys, (key) => {
-                const resetValue = lodashGet(defaultKeyStates, key, null);
-                cache.set(key, resetValue);
-                notifySubscribersOnNextTick(key, resetValue);
+            const defaultKeys = _.keys(defaultKeyStates);
+            const keysToClear = _.without(keys, ..._.union(keysToPreserve, defaultKeys));
+            const keyValuesToClear = _.map(keysToClear, key => [key, null]);
+            const defaultKeyValues = _.pairs(_.omit(defaultKeyStates, ...keysToPreserve));
+            const newStoreKeyValues = _.union(keyValuesToClear, defaultKeyValues);
```

Comment:
Thanks for the additional comments! I think what is still confusing is some of the variable names.

- `keyValuesToClear` sounds like it would be the thing that eventually is passed to `multiSet()` just by the name of it
- `newStoreKeyValues` is what actually gets passed to `multiSet()` but the name doesn't really imply that at all. "New" is pretty meaningless in the name of this variable

What I would suggest is to start with an empty variable:

```
const keyValuesToClear = [];
```

Then as you progress through the logic, things will get added to that array.

```
keyValuesToClear.push(defaultKeyValues);
```

Eventually, that's what gets passed to multiSet

```
Storage.multiSet(keyValuesToClear);
```